internal/pipe/sbom/sbom.go

```diff
@@ -0,0 +1,222 @@
+package sbom
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/internal/gio"
+	"github.com/goreleaser/goreleaser/internal/ids"
+	"github.com/goreleaser/goreleaser/internal/logext"
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+// Pipe that catalogs common artifacts as an SBOM.
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "cataloging artifacts" }
+func (Pipe) Skip(ctx *context.Context) bool { return len(ctx.Config.SBOMs) == 0 }
+
+// Default sets the Pipes defaults.
+func (Pipe) Default(ctx *context.Context) error {
+	ids := ids.New("sboms")
+	for i := range ctx.Config.SBOMs {
+		cfg := &ctx.Config.SBOMs[i]
+		if cfg.Cmd == "" {
+			cfg.Cmd = "syft"
+			//return errors.New("cataloging artifacts failed: no command specified")
+		}
+		if len(cfg.SBOMs) == 0 {
+			var sbom string
+			switch cfg.Artifacts {
+			case "binary":
+				sbom = "{{ .ArtifactName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}.sbom"
+			default:
+				sbom = "{{ .ArtifactName }}.sbom"
+			}
+			cfg.SBOMs = []string{sbom}
+		}
+		if len(cfg.Args) == 0 {
+			cfg.Args = []string{"--file", "{{ .Dist }}/$sbom0", "--output", "spdx-json", "$artifact"}
```

Comment:
on the sign pipe prefix the file with the dist when setting the env... I would say we follow the same path here so both usages/impls look similar...